src/chatlistmodel.cpp

```diff
@@ -795,31 +761,15 @@ void ChatListModel::handleChatPhotoUpdated(qlonglong chatId, const QVariantMap &
         ChatData *chat = hiddenChats.value(chatId);
         if (chat) {
             LOG("Updating photo for hidden chat" << chatId);
-            chat->chatData.insert(PHOTO, photo);
-        }
-    }
-}
-
-void ChatListModel::handleChatPinnedMessageUpdated(qlonglong chatId, qlonglong pinnedMessageId)
-{
-    if (chatIndexMap.contains(chatId)) {
-        LOG("Updating pinned message for" << chatId);
-        const int chatIndex = chatIndexMap.value(chatId);
-        ChatData *chat = chatList.at(chatIndex);
-        chat->chatData.insert(PINNED_MESSAGE_ID, pinnedMessageId);
-    } else {
-        ChatData *chat = hiddenChats.value(chatId);
-        if (chat) {
-            LOG("Updating pinned message for hidden chat" << chatId);
-            chat->chatData.insert(PINNED_MESSAGE_ID, pinnedMessageId);
+            chat->photoSmall = photo.value(SMALL);
         }
     }
 }
 
 void ChatListModel::handleMessageSendSucceeded(qlonglong messageId, qlonglong oldMessageId, const QVariantMap &message)
 {
     bool ok;
-    const qlonglong chatId(message.value(CHAT_ID).toLongLong(&ok));
+    qlonglong chatId(message.value(CHAT_ID).toLongLong(&ok));
```

Comment:
I don't like `this->`, somebody doesn't like `const` :)

I would like to point out though, that unlike `this->`, `const` is functional. It a) makes sure that you don't update a variable which is not supposed to be updated and b) gives the compiler a chance to better optimize the code.

Reply:
Oh yeah, I stumbled a bit there. 
This was mainly to get rid of warnings where I used it in the following lines. If we end up actually using this, I'll be sure to to adress this. Thanks for the explanation!

Reply:
Not to mention that marking an instance of a Qt collection (QMap, QList and such) as const makes sure that non-const methods don't get accidentally invoked and shared content doesn't get unnecessarily copied (or "detached" as it's sometimes called).

Reply:
Huh, the warnings I was talking about may already have been resolved by changing other things in between. Didn't get any after just putting it in. Nice.